cranelift-codegen/src/binemit/stackmap.rs

```diff
@@ -0,0 +1,124 @@
+use crate::bitset::BitSet;
+use crate::ir;
+use crate::isa::TargetIsa;
+use std::vec::Vec;
+
+/// Wrapper class for longer bit vectors that cannot be represented by a single BitSet
+#[derive(Clone, Debug)]
+pub struct Stackmap {
+    bitmap: Vec<BitSet<u32>>,
+}
+
+impl Stackmap {
+    /// Create a stackmap based on where references are located on a function's stack
+    pub fn from_values(
+        args: &[ir::entities::Value],
+        func: &ir::Function,
+        isa: &dyn TargetIsa,
+    ) -> Self {
+        let stack = &func.stack_slots;
+        let loc = &func.locations;
+        let mut live_ref_in_stack_slot: std::collections::HashSet<ir::StackSlot> =
+            std::collections::HashSet::new();
+        // References can be in registers, but live registers values are pushed onto the stack before calls and traps.
```

Comment:
I'm not sure what you're referring to, so maybe @julian-seward1 can help answer this.

As far as this change is concerned, I'm imitating the way `Stackmap`s are constructed in Spidermonkey -- which focuses only on the stack. The reason seems to be that registers are already pushed onto the stack by the time we construct the `Stackmap`. Is that also true for Cranelift?

Reply:
I've discussed with Julian in private. My main worry that if a register contained a Ref, and it was spilled but reused after a safepoint, it could contain some stall ref value (if the GC decided to relocate the value, e.g. in the case of a generational GC).

In Spidermonkey, it is safe to do so because we have our own invariants that wasm stack maps don't need to worry about registers at all. In general for Cranelift, I don't think it's the case, and later (not in this PR), we should implement register maps too. So the implementation is perfect for our needs here. Can you add a comment that we should maybe worry about registers later, please? Thanks!